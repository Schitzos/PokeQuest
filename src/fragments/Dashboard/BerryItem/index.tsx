import React from 'react';
import TextView from '@/components/TextView';
import {StyleSheet, TouchableOpacity} from 'react-native';
import theme from '@/theme';
import FastImage from 'react-native-fast-image';

export interface BerryItemProps {
  name: string;
  handleSelectBerry: (data: any) => void;
  selectedBerry: any;
}

export interface BerryItemDataResponse {
  name: any;
  item: {
    url: string;
  };
}

export interface BerryItemDetailDataResponse {
  firmness: {
    name: string;
  };
  id: number;
}

export default function BerryItem({
  name,
  handleSelectBerry,
  selectedBerry,
}: BerryItemProps) {
  const isSelected = selectedBerry === name;

  return (
    <TouchableOpacity
      style={[styles.base, isSelected && styles.selectedBase]}
      onPress={() => handleSelectBerry(name)}>
      <FastImage
        style={styles.berryArt}
        source={{
          uri: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/items/${name}-berry.png`,
        }}
      />
      <TextView fz={10}>{name}</TextView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  base: {
    width: '33%',
    alignItems: 'center',
    paddingVertical: 4,
  },
  selectedBase: {
    borderColor: theme.colors.primary,
    borderWidth: 1,
    borderRadius: 8,
  },
  berryArt: {
    width: 16,
    height: 16,
  },
});
