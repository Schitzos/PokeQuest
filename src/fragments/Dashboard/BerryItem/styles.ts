import theme from '@/theme';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  base: {
    width: '33%',
    alignItems: 'center',
    paddingVertical: 4,
  },
  selectedBase: {
    borderColor: theme.colors.primary600,
    borderWidth: 1,
    borderRadius: 8,
    backgroundColor: theme.colors.neutral50,
  },
  berryArt: {
    width: 16,
    height: 16,
  },
});
