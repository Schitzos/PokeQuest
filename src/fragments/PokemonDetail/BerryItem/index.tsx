import React from 'react';
import TextView from '@/components/TextView';
import {
  getBerryDetail,
  getBerryItemDetail,
} from '@/services/berry/berry.service';
import {Image, StyleSheet, TouchableOpacity} from 'react-native';

export interface BerryItemProps {
  name: any;
}

export interface BerryItemDataResponse {
  name: any;
  item: {
    url: string;
  };
}

export interface BerryItemDetailDataResponse {
  sprites: {
    default: string;
  };
}

export default function BerryItem({name}: BerryItemProps) {
  const berryDetail = getBerryDetail({
    key: ['getBerryDetail', name],
    name: name,
  });

  const berry = (berryDetail.data || {}) as BerryItemDataResponse;
  const berryId = berry?.item?.url?.split('/').slice(-2, -1)[0];

  const berryItemDetail = getBerryItemDetail({
    key: ['getBerryItemDetail', name],
    id: berryId,
  });

  const itemDetail = (berryItemDetail.data ||
    {}) as BerryItemDetailDataResponse;
  const itemImage = itemDetail?.sprites?.default || '';

  return (
    <TouchableOpacity style={styles.base}>
      {itemImage && (
        <Image
          style={styles.artwork}
          source={{
            uri: itemImage,
          }}
        />
      )}
      <TextView>{berry.name}</TextView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  base: {
    width: '20%',
    height: 25,
    alignItems: 'center',
  },
  artwork: {
    width: 24,
    height: 24,
  },
});
