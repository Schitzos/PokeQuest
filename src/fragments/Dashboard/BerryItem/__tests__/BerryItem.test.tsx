import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import BerryItem from '../index';

describe('BerryItem component', () => {
  const mockHandleSelectBerry = jest.fn();

  test('calls handleSelectBerry function when pressed', () => {
    const {getByText} = render(
      <BerryItem
        name="testBerry"
        handleSelectBerry={mockHandleSelectBerry}
        selectedBerry=""
      />,
    );
    const berry = getByText('testBerry');
    fireEvent.press(berry);
    expect(mockHandleSelectBerry).toHaveBeenCalledWith('testBerry');
  });

  test('matches snapshot', () => {
    const {toJSON} = render(
      <BerryItem
        name="testBerry"
        handleSelectBerry={mockHandleSelectBerry}
        selectedBerry=""
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
