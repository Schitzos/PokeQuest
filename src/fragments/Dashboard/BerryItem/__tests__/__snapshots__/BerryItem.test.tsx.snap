// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerryItem component matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "opacity": 1,
      "paddingVertical": 4,
      "width": "33%",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 16,
          "width": 16,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={
        {
          "testUri": "../../../assets/images/default_image_loading.png",
        }
      }
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#333",
          "fontFamily": "PixeloidMono",
          "fontSize": 10,
          "fontWeight": "normal",
          "textAlign": "left",
          "textTransform": "capitalize",
        },
      ]
    }
  >
    testBerry
  </Text>
</View>
`;
